articles/rock-paper-scissor-online/index.md

```diff
@@ -0,0 +1,248 @@
+### Listen to Player's Choice
+We add server-side side logic at `app.js` to listen to Player1's choice.
+```
+//Listener to Player 1's Choice
+    socket.on("choice1", (data)=> {
+        choice1 = data.choice;
+        console.log(choice1, choice2);
+        if (choice2 != "") {
+            result(data.roomID);
+        }
+    });
+```
+
+The above code gets player1's choice and does nothing if player2 hasn't picked their choice yet.
+
+```
+//Listener to Player 2's Choice
+    socket.on("choice2", (data)=> {
+        choice2 = data.choice;
+        console.log(choice1, choice2);
+        if (choice1 != "") {
+            result(data.roomID);
+        }
+    });
+```
+Next we write logic to listen to player2's choice in `app.js`. Once, both of player have picked their choice, the server enters into the `if block` and invokes the `result()` function.
+
+
+
+### Declare Winner
+
+In `app.js`,
+```
+//Function to be executed after getting both choices
+const result=(roomID)=> {
+    var winner = getWinner(choice1, choice2);
+    io.sockets.to(roomID).emit("result", {
+        winner: winner
+    });
+    choice1 = "";
+    choice2 = "";
+}
+```
+The result function takes the roomID as an argument. First, the `getWinner()` function calculates the result based on the player's choices .   Next, it emits the result to all the clients in the room, i.e both player1 and player2. Finally, it resets the player's choices.
+
+> Note: `getWinner()` is already included in the boiler code.
+
+### Listen to Result
+
+We now have the server emitting the result to both the players. It's time to write client-side logic at `public/game.js` to listen to the `result` event.
+
+```
+//Result Event Listener
+socket.on("result",(data)=>{
+    if(data.winner=="draw"){
+        $("#message").html("It's a draw!");
+    }else{
+        updateDOM(firstPlayer==data.winner?"player1":"player2");
+    }
+})
+
+const updateDOM=(player)=>{
+    const playerDOM=$("."+player+" span");
+    const prevScore=parseInt(playerDOM.html().trim());
+    playerDOM.html(prevScore+1);
+    const winnerName=$("."+player+" .name").html().trim();
+    $("#message").html(winnerName+" scored a point!");
+}
+```
+
+The above code listens to the result event and updates the DOM with the help of `updateDOM()` function. The `updateDOM()` function announces the winner and increases the score of the winner. In case of a draw, we just display a message that it's a draw.
+
+### Conclusion
+
+Kudos, You made it 🎉
+
+This app although works, yet could use a lot of improvement. As mentioned before, the intention was to keep it simple to put more focus Socket logic. Learning never stops here, now that the app works the next step is to do improvements in order to have a fruitful experience.
```

Comment:
**Flow Edit:** Although the app works, it 

add **on** after focus

I'd rephrase the entire learning never stops sentence. Something like: The next step is to improve the app by adding extra functionality. Some suggestions are **(suggest some additions to readers)**

Reply:
The new structure looks like,

...
The next step is to improve the app by adding extra functionality, read along for some suggestions.

### Next Steps
...